src/index.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+const assert = require('assert');
+const {Client} = require('remotely-signed-s3');
+
+class Artifact {
+  constructor({queue, client, clientOpts}) {
+    assert(typeof queue === 'object', 'Must provide a Queue client');
+    this.__queue = queue;
+
+    if (client && clientOpts) {
+      throw new Error('You can only provide a client or options for a client to be created with');
```

Comment:
`You can only provide a client or options for a client be created with` doesn't sound right to me.